packages/backend/server/src/core/doc-renderer/controller.ts

```diff
@@ -214,11 +222,24 @@ export class DocRendererController {
    */
   private readHtmlAssets(path: string): HtmlAssets {
     const manifestPath = join(path, 'assets-manifest.json');
-    const htmlPath = join(path, 'index.html');
 
     try {
-      const assets = JSON.parse(readFileSync(manifestPath, 'utf-8'));
-      assets.html = readFileSync(htmlPath, 'utf-8');
+      const assets: HtmlAssets = JSON.parse(
+        readFileSync(manifestPath, 'utf-8')
+      );
+
+      const publicPath = this.config.isSelfhosted
+        ? this.config.server.host + assets.selfhostPublicPath
+        : assets.publicPath;
+
+      if (this.config.isSelfhosted) {
+        // append `config.server.host` to public path
+        assets.selfhostPublicPath = publicPath;
+      }
+
+      assets.js = assets.js.map(path => publicPath + path);
```

Comment:
if `publicPath` is `/`, and the path is `/js/main.js`, the result is `//js/main.js`, this is not expected, seems should remove the prefix `/` of path